test/jdk/javax/swing/JTable/JTableScrollPrintTest.java

```diff
@@ -0,0 +1,203 @@
+                add(table);
+            }
+        }
+    }
+
+    static class PrintUtilities implements Printable {
+        private Component componentToBePrinted;
+        public void printComponent(Component c, String jobname) {
+            new PrintUtilities(c).print(jobname);
+        }
+
+        public PrintUtilities(Component componentToBePrinted) {
+            this.componentToBePrinted = componentToBePrinted;
+        }
+
+        public void print(String jobname) {
+            PrinterJob printJob = PrinterJob.getPrinterJob();
+            PageFormat pf = printJob.defaultPage();
+            pf.setOrientation(PageFormat.PORTRAIT);
+
+            // set margins to 1/2"
+            Paper p = new Paper();
+            p.setImageableArea(36, 36, p.getWidth()-72, p.getHeight()-72);
+            pf.setPaper(p);
+
+            printJob.setPrintable(this, pf);
+            printJob.setJobName(jobname);
+
+            if (printJob.printDialog()) {
+                try {
+                    printJob.print();
+                } catch(PrinterException pe) {
+                    System.out.println("Error printing: " + pe);
+                }
+            }
+        }
+
+        public int print(Graphics g, PageFormat pageFormat, int pageIndex) {
+            if (pageIndex > 0) {
+                return(NO_SUCH_PAGE);
+            } else {
+                Graphics2D g2d = (Graphics2D)g;
+                g2d.translate(pageFormat.getImageableX(), pageFormat.getImageableY());
+                Component c= componentToBePrinted;
+                c.setSize(c.getPreferredSize());
+
+                double panelX= c.getWidth();
+                double panelY= c.getHeight();
+                float imageableX = (float) pageFormat.getImageableWidth()-1;
+                float imageableY = (float) pageFormat.getImageableHeight()-1;
+
+                double xscale = imageableX/panelX;
+                double yscale = imageableY/panelY;
+                double optimalScale;
+                if (xscale < yscale)
+                    optimalScale = xscale;
+                else
+                    optimalScale = yscale;
+
+                if (optimalScale > 1)
+                    optimalScale = 1;
+
+                g2d.scale(optimalScale, optimalScale);
+                disableDoubleBuffering(c);
+                c.paint(g2d);
+                enableDoubleBuffering(c);
+                return(PAGE_EXISTS);
+            }
+        }
+
+        public void disableDoubleBuffering(Component c) {
+            RepaintManager currentManager = RepaintManager.currentManager(c);
+            currentManager.setDoubleBufferingEnabled(false);
+        }
+
+        public void enableDoubleBuffering(Component c) {
+            RepaintManager currentManager = RepaintManager.currentManager(c);
+            currentManager.setDoubleBufferingEnabled(true);
+        }
+    }
+
+    protected void processWindowEvent(WindowEvent e) {
```

Comment:
I guess this is not required

Also, please confirm JTable is  getting printed correctly either from existing testcases in javax/swing/JTable and from SwingSet2..

Reply:
Yes, verified with JTable print test cases and SwingSet2.